helix-term/src/ui/editor.rs

```diff
@@ -1062,6 +1062,17 @@ impl Component for EditorView {
                     (Mode::Insert, Mode::Normal) => {
                         // if exiting insert mode, remove completion
                         self.completion = None;
+
+                        let last_cmd = self.last_insert.0;
+                        // For user friendly,
+                        // Remove whitespaces if we go from insert mode(through open below/above) to normal mode without any keys in between.
+                        // Example: `o<esc>`.
+                        if (last_cmd.name() == "open_below" || last_cmd.name() == "open_above")
+                            && self.last_insert.1.len() == 1
+                        {
```

Comment:
Add a `fun()` method then match on the function directly. This way it'll be a compile error if the function names change, and you're comparing pointers, not strings.

```suggestion
                        if matches!(commands::open_below | commands::open_above, last_cmd.fun())
                            && self.last_insert.1.len() == 1
                        {
```

Also, shouldn't `last_insert.1.len()` be empty?

Reply:
Done, it's harder than I think, it seems that I can't use match for fuction directly.  So I create a new function type for command func, and two const variable(open_below and open_above) to handle this.  Here is the relative issue:
https://github.com/rust-lang/rust/issues/44333

Or, if we have a better way to handle this?

> Also, shouldn't last_insert.1.len() be empty?

It's not empty, because here we always insert key first, and then check mode transaction, the `last_insert` only be clear when goes from `Normal mode` to `Insert mode`.

https://github.com/helix-editor/helix/blob/67bf4250caf14d3a632abdbfa367c04b318d782c/helix-term/src/ui/editor.rs#L990-L992

Reply:
>  it seems that I can't use match for fuction directly

A simple `if` should still work though.

Reply:
Hmm.... I have tried this
```rust
if (last_cmd == commands::open_below) {
}
```

But compiler throw out an error:
```
error[E0369]: binary operation `==` cannot be applied to type `for<'r, 's> fn(&'r mut commands::Context<'s>)`
```